### PR TITLE
add new imagepicker

### PR DESCRIPTION
**Pull request by sleushunou**

### CODE REVIEW

File: Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/DroidImagePickerResult.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            Stream memoryStream = new MemoryStream();
+            if(ImageObject != null)
+            {
+                if(ImageExtension == ImageExtension.Jpg)
+                {
+                    ((Bitmap) ImageObject).Compress(CompressFormat.Jpeg, (int)(Quality * 100), memoryStream);
+                }
+                else if (ImageExtension == ImageExtension.Png)
+                {
+                    ((Bitmap) ImageObject).Compress(CompressFormat.Png, (int) (Quality * 100), memoryStream);
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/DroidImagePickerResult.cs]: duplicated lines differences only enum value

File: Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+using Android.Runtime;
+using Android.Support.V4.Content;
+using Android.Widget;
+using Plugin.CurrentActivity;
+using ImageOrientation = Android.Media.Orientation;
+
+namespace Softeq.XToolkit.WhiteLabel.Droid.ImagePicker
+{
+    [Activity(Label = "ImagePickerActivity")]
+    internal class ImagePickerActivity : Activity
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs]: need to discuss, that we really want to support our activity or use media plugin activity

File: Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+using Android.Provider;
+using Android.Runtime;
+using Android.Support.V4.Content;
+using Android.Widget;
+using Plugin.CurrentActivity;
+using ImageOrientation = Android.Media.Orientation;
+
+namespace Softeq.XToolkit.WhiteLabel.Droid.ImagePicker
+{
+    [Activity(Label = "ImagePickerActivity")]
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs]: remove Label

File: Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+        {
+            subdir = subdir ?? string.Empty;
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                var timestamp = DateTime.Now.ToString("yyyyMMdd_HHmmss", CultureInfo.InvariantCulture);
+                name = "IMG_" + timestamp + ".jpg";
+            }
+
+            var mediaType = Android.OS.Environment.DirectoryPictures;
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs]: need to discuss where we save an image:
- in global user pictures
- in external for app (when removing)

File: Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/DroidImagePicker.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+using Plugin.Permissions;
+using Softeq.XToolkit.Permissions;
+using Softeq.XToolkit.WhiteLabel.ImagePicker;
+
+namespace Softeq.XToolkit.WhiteLabel.Droid.ImagePicker
+{
+    public class DroidImagePicker : IImagePicker
+    {
+        private readonly IPermissionsManager _permissionsManager;
+        private const string TAG_PIXEL_X_DIMENSION = "PixelXDimension";
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/DroidImagePicker.cs]: rudimental consts

File: Softeq.XToolkit.WhiteLabel/ImagePicker/ImagePickerResult.cs [hunk 6]
```diff
@@ -53,6 +53,45 @@ public enum ImageExtension
         Jpg
     }
 
+    public abstract class ImagePickeResult : IDisposable
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel/ImagePicker/ImagePickerResult.cs]: typo `ImagePickerResult `

File: Softeq.XToolkit.WhiteLabel/ImagePicker/IImagePicker.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Softeq.XToolkit.WhiteLabel.ImagePicker
+{
+    public interface IImagePicker
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel/ImagePicker/IImagePicker.cs]: what do you think about rename to `IImagePickerService`?

File: Softeq.XToolkit.WhiteLabel.iOS/ImagePicker/IosImagePickerResult.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+using Softeq.XToolkit.WhiteLabel.ImagePicker;
+using UIKit;
+
+namespace Softeq.XToolkit.WhiteLabel.iOS.ImagePicker
+{
+    public class IosImagePickerResult : ImagePickeResult
+    {
+        public override Task<Stream> GetStream()
+        {
+            if(ImageObject != null)
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.iOS/ImagePicker/IosImagePickerResult.cs]: can use pattern matching:
if (ImageObject is UIImage image) {}

File: Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs (hunk 4 above)
Comment (wcoder) [reply Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs]: After discussed with @sleushunou we will use cache

File: Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+        private System.IO.Stream GetContentSrream(Context context, Android.Net.Uri uri)
+        {
+            var stream = System.IO.Stream.Null;
+            try
+            {
+                stream = context.ContentResolver.OpenInputStream(uri);
+            }
+            catch (Java.IO.FileNotFoundException fnfEx)
+            {
+                System.Diagnostics.Debug.WriteLine("Unable to save picked file from disk " + fnfEx);
```
Comment (fryette) [inline Softeq.XToolkit.WhiteLabel.Droid/ImagePicker/ImagePickerActivity.cs]: import static usings

**Pull request merged**